sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -343,6 +343,31 @@ def delete_container(
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace
+    def undelete_container(self, deleted_container_name, deleted_container_version, **kwargs):
```

Comment:
The ContainerClient already represents a single named container. So we should remove this parameter and use the clients container_name directly.
If we must pass in this parameter - then this operation should move up to the parent client.

Reply:
deleted_container_name is the name when it was deleted, container_name is the one you want to restore to, those two names could be different